src/main/service/listManipulation/OpaPressListService.ts

```diff
@@ -13,32 +13,33 @@ export class OpaPressListService {
                 courtRoom.session.forEach(session => {
                     session.sittings.forEach(sitting => {
                         sitting.hearing.forEach(hearing => {
-                            const defendantInfo = this.processPartyRoles(hearing);
-
-                            // Each case can have multiple defendants. They will be shown as separate entry on the list
-                            // as each can have its own plea date
-                            defendantInfo.forEach(defendant => {
-                                const rows = [];
-                                hearing.case.forEach(hearingCase => {
-                                    const caseInfo = this.buildHearingCase(hearingCase);
-                                    const row = { ...caseInfo, ...defendant };
-                                    rows.push(row);
+                            hearing.case.forEach(hearingCase => {
+                                const defendantInfo = this.processPartyRoles(hearingCase);
+
+                                // Each case can have multiple defendants. They will be shown as separate entry on the list
+                                // as each can have its own plea date
+                                defendantInfo.forEach(defendant => {
+                                    const rows = [];
+                                    hearing.case.forEach(hearingCase => {
```

Comment:
Found a bug here during testing. It seems to be duplicating parties, and I believe this is down to this line.

At line 16, it is looping through each case, therefore line 23 is no longer needed as it is looping through the cases a second time. This is then causing a cartesian product multiplication, where parties and cases are multiplied together.

A similar thing appears to be happening in channel management too, however I have not dug into the exact reason why (I imaging similar as it is causing a similar result)

Reply:
Modified OpaPressListService.ts, and ran OpaPressListService.ts test, which was successful.